samples/wifi/monitor/src/main.c

```diff
@@ -328,6 +328,57 @@ static int wifi_set_channel(void)
 	return 0;
 }
 
+static int wifi_set_packet_filter(void)
+{
+	int ret;
+	struct net_if *iface;
+	struct wifi_filter_info filter_info = { 0 };
+	uint32_t filter = 0;
+
+	iface = net_if_get_first_wifi();
+	if (iface == NULL) {
+		LOG_ERR("No Wi-Fi interface found");
+		return -1;
+	}
+
+	filter_info.if_index = net_if_get_by_iface(iface);
+
+#if defined(CONFIG_MONITOR_MODE_WIFI_PACKET_FILTER_ALL)
```

Comment:
```suggestion
   if (IS_ENABLED(CONFIG_MONITOR_MODE_WIFI_PACKET_FILTER_ALL))
```
We don't care about memory here, but want to catch any build errors. For below as well, move conditionals to runtime.